dozer-sql/src/pipeline/product/set.rs

```diff
@@ -45,14 +48,69 @@ impl SetOperation {
         &self,
         action: SetAction,
         record: &Record,
-        record_hash_map: &mut Vec<u64>,
+        database: &Database,
+        txn: &SharedTransaction,
     ) -> Result<Vec<(SetAction, Record)>, PipelineError> {
-        let mut output_records: Vec<(SetAction, Record)> = vec![];
-        let lookup_key = record.get_values_hash();
-        if !record_hash_map.contains(&lookup_key) {
-            output_records.push((action, record.clone()));
-            record_hash_map.push(lookup_key);
+        let lookup_key = record.get_values_hash().to_be_bytes();
+        let write_txn = &mut txn.write();
+        match action {
+            SetAction::Insert => {
+                self.update_set_db(&lookup_key, 1, false, write_txn, *database);
+            }
+            SetAction::Delete => {
+                self.update_set_db(&lookup_key, 1, true, write_txn, *database);
+            }
+        }
+        let curr_count = self.get_occurrence(&lookup_key, write_txn, *database)?;
```

Comment:
can't we use the `new_count` value that has just been written with `update_set_db`, rather then reading it with `get_occurrence`?

Reply:
ok - this was for double confirmation that we have recorded correctly